integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/CarbonDropTableCommand.scala

```diff
@@ -69,7 +69,7 @@ case class CarbonDropTableCommand(
                 CarbonLockUtil.getLockObject(identifier, lock)
       }
       // check for directly drop datamap table
-      if (carbonTable.isChildTable && !dropChildTable) {
+      if (carbonTable.isChildTableForMV && !dropChildTable) {
```

Comment:
This file has a big section for pre-aggregate, we can remove that code

` if (carbonTable.getTableInfo.getParentRelationIdentifiers.size() == 1) {`

Reply:
fixed